exec/java-exec/src/main/java/org/apache/drill/exec/planner/logical/DrillFilterAggregateTransposeRule.java

```diff
@@ -0,0 +1,28 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.exec.planner.logical;
+
+import org.apache.calcite.rel.rules.FilterAggregateTransposeRule;
+
+public class DrillFilterAggregateTransposeRule{
```

Comment:
The customized rule is not only adding a customer filter factory, it also specifies the filter / aggregate has to be in Drill Logical.  The default rule patten will match Filter and Aggregate [1]. In debug, we saw that the default one will match with a LogicalFilter on top of DrillAggregate, which I feel is not right. We do not want to max a Rels with mixed conventions.

@sudheeshkatkam helped track to the following error, which seems indicate the rule matching a LogicalFilter /DrillAggregate. Then, exception handling seems hit a cyclic somehow, and hang there forever. 

The intention of this customized rule is to ensure that we do not match rel nodes with mixed convention.


Internal error: Error while applying rule FilterAggregateTransposeRule, args [rel#217:LogicalFilter.NONE.ANY([]).[[]](input=rel#126:Subset#5.LOGICAL.ANY([]).[],condition=IS NOT NULL($1)), rel#6972:DrillAggregateRel.LOGICAL.ANY([]).[](input=rel#6971:Subset#1714.NONE.ANY([]).[],group={0, 1})]

 
[1] https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/rel/rules/FilterAggregateTransposeRule.java#L52-L53